storage/database.go

```diff
@@ -490,13 +485,20 @@ func (d *db) splayedTick() {
 		time.Sleep(d.tickDeadline - duration)
 	}
 
-	// NB(r): Cleanup and/or flush if required to cleanup files and/or
-	// flush blocks to disk. Note this has to run after the tick as
-	// blocks may only have just become available during a tick beginning
-	// from the tick begin marker.
-	if d.fsm.ShouldRun(start) {
-		d.fsm.Run(start, true)
-	}
+	go func() {
+		// NB(r): Cleanup and/or flush if required to cleanup files and/or
+		// flush blocks to disk. Note this has to run after the tick as
+		// blocks may only have just become available during a tick beginning
+		// from the tick begin marker.
+		if d.fsm.ShouldRun(start) {
+			d.fsm.Run(start, false)
+		}
+
+		// NB(prateek): The repairer has to be run after any pending flushes
+		if d.repairer.ShouldRun() {
```

Comment:
I'm thinking we should probably launch these as two separate goroutines so they don't affect each other.  Is this ok?

I really don't want to see flushing delayed 30-60mins because the last repair took extra long to run.